### PR TITLE
chore(deps): bump Talos to v1.14.0

### PR DESCRIPTION
**Pull request by sircthulhu**
## Summary

Tracks upstream Talos v1.14.0 for the main module and `pkg/machinery`, and fixes what the bump broke.

| Dependency | Before | After |
| --- | --- | --- |
| `github.com/siderolabs/talos` | v1.13.7 | v1.14.0 |
| `github.com/siderolabs/talos/pkg/machinery` | v1.13.7 | v1.14.0 |
| `k8s.io/{api,apimachinery,client-go,component-base}` | v0.36.2 | v0.37.0 |

## Breaking: two Chart.yaml keys stop working when empty

A project that left `templateOptions.talosVersion` and `templateOptions.kubernetesVersion` unset will not render until both are pinned.

Empty `talosVersion` still means the Talos version talm was built against. From v1.14 that contract keeps the Kubernetes settings in documents of their own, while the charts write them as v1alpha1 fields, and machinery rejects a config carrying both. So an unpinned project was rendering something the node refuses, and nothing said so until the apply failed. The render stops now and names the fields.

Empty `kubernetesVersion` used to mean the node picks: machinery emitted no image field. v1.14 will not generate without a version, and the fallback that satisfies it writes talm's own Kubernetes version into the config, where the node file does not show it. Below that contract the generated images are stripped and the node still picks. On it, the missing pin is reported with the key to set.

`engine.SerializeConfiguration` takes the two version strings now, so its signature changed. It is exported, and the contract it had before (serialize whatever the bundle holds) could not survive the version handling above.

Both shipped presets pin both keys now, so generic-preset projects will report preset drift until re-synced. `docs/operations/upgrading.md` has the migration, including why taking the preset's `kubernetesVersion` without checking can move a control plane in the wrong direction.

## Also fixed

`talm init` stopped working entirely: `config/generate` errors on an empty Kubernetes version instead of defaulting it.

A full render kept 1 document out of 28. The CA and the service-account key were among the ones dropped.

`talosctl meta --insecure` became unreachable upstream: the flag moved onto the container command as a local flag, which reaches neither the subcommands nor the parent. siderolabs/talos#14347 fixes it but has not shipped in a v1.14 release, so talm re-publishes the flag itself, marked for removal.

Pre-apply link checks cover `VethConfig` and `BGPInstanceConfig`, and the host snapshot now carries link aliases and altnames. Machinery takes an alias anywhere it takes a link name, so a bond over an aliased NIC was blocked on a config Talos accepts. That one predates this branch.

## MPL-2.0

v1.14 dropped `helpers.FailIfMultiNodes`, `helpers.ForEachResource`, `global.Args.WithClientNoNodes` and `global.Args.WithClientMaintenance` with no exported replacement. talm carries ports in `pkg/engine/talos_helpers.go` and `pkg/commands/talos_client.go`. Talos is MPL-2.0, so both files carry the MPL notice; the rest of talm stays Apache-2.0 under MPL-2.0 section 3.3.

## Testing

`go build ./...`, `go test ./...` and `golangci-lint run` clean locally. New contract tests load a rendered config back through machinery — nothing in the suite did that before, so a render could stop being applicable while everything stayed green. On-cluster steps are in `docs/manual-test-plan.md`.

## Summary by CodeRabbit

- **New Features**
  - Added Talos v1.14 compatibility, multi-document rendering, and version pinning.
  - Added strict chart and preset drift checks for release builds.
  - Added certificate fingerprint support, container namespace selection, and encrypted support bundles.
  - Added validation for link aliases, alternate names, VRF, WireGuard, Veth, and BGP references.
  - Added documentation for LVM and software RAID wipe commands.

- **Bug Fixes**
  - Fixed nested Talos command flag propagation and configuration serialization.
  - Improved Kubernetes version handling and certificate fingerprint guidance.

- **Documentation**
  - Expanded upgrade, compatibility, safety, command-reference, and licensing guidance.
  - Documented removal of unsupported insecure and reboot modes.

### CODE REVIEW

**Review by lexfrei — DISMISSED** [review]
NOT LGTM — new TLS-skip / version-fallback logic ships without tests, and the bump pins a production bootstrap tool to a Talos alpha; both need resolution before merge.

**Business context**: bump Talos to v1.14 machinery so talm recognizes the new v1.14-only `KubeEtcdEncryptionConfig` document, dropping the cozystack/talos fork that previously carried `--skip-verify`.

What was verified and is **not** a problem (raising so reviewers don't re-litigate):

- **Fork patches preserved.** The fork at the pinned commit was upstream v1.12.1 plus exactly two commits: the rotate-ca `--k8s-endpoint` fix (independently present in upstream v1.14.0-alpha.1 — `rotate-ca.go:153`, `pkg/rotate/pki/kubernetes/kubernetes.go:46/176`) and `--skip-verify` (faithfully reimplemented in `pkg/commands/root.go`). No carried patch is lost.
- **Scope is cohesive.** The skip-verify reimplementation, context threading, `k8s.io` bump, and explicit-Kubernetes-version handling are all forced consequences of the v1.14 client-API and `config/generate` changes, not unrelated additions.

## Blockers

### B1: Version choice — pinning to a Talos alpha

**File**: `go.mod:82,122`

**Issue**: This pins talm (a production bootstrap tool) to v1.14.0-alpha.1, a month-old prerelease, when stable v1.13.5 exists.

**Evidence**: The sole capability v1.14 adds over v1.13.x is the `KubeEtcdEncryptionConfig` type — confirmed present only in v1.14.0-alpha.1 (absent in v1.13.5 / v1.13.3). On every other axis stable is ahead: v1.13.5's `DefaultKubernetesVersion` is `1.36.2` vs the alpha's `1.36.1`, and its config schema is frozen while the alpha's `KubeEtcdEncryptionConfig` can still change before v1.14.0 ships.

**Impact**: The platform would track an unreleased Talos with a non-final machine-config schema and an older default Kubernetes version than current stable.

**Decision needed**: Is `KubeEtcdEncryptionConfig` required before v1.14.0 stable releases? If not, v1.13.5 is the better target. If yes, please state that justification in the PR body so the alpha pin is a deliberate, documented tradeoff.

### B2: New logic ships without tests

**File**: `pkg/commands/root.go` (`skipVerifyTLSConfig`, `WithClientSkipVerify`), `pkg/engine/engine.go` (`kubeVersion`)

**Issue**: The reimplemented `--skip-verify` path and the new `kubeVersion` fallback have no unit tests; no existing test references them.

**Evidence**: grep for `skipVerifyTLSConfig` / `kubeVersion` / `WithClientSkipVerify` across `*_test.go` returns nothing, despite 33 test files in `pkg/commands`. `go test ./...` and the coverage CI job passing only confirm existing tests still pass — they do not cover the added lines.

**Impact**: `skipVerifyTLSConfig` disables TLS verification and parses the client cert/key (three error paths); `kubeVersion` encodes the "fall back to default when version unset" guarantee the PR body promises to preserve. Both are load-bearing and currently unverified.

**Fix**: Add table tests for `kubeVersion` (empty -> `DefaultKubernetesVersion`, `"v1.30.0"` -> `"1.30.0"`) and for `skipVerifyTLSConfig` (cert+key present, both absent, malformed base64, nil/empty context). The context-resolution and `errContextNotFound` paths in `WithClientSkipVerify` are also worth a test.

## Non-blocking follow-ups

1. `pkg/commands/root.go`: `cfg.Contexts[contextName]` can return `(nil, true)` if the talosconfig defines a context as null; the later `configContext.Crt` deref would panic. Carried over from the fork, low severity (needs a malformed config), but a `|| configContext == nil` guard is cheap — and a `skipVerifyTLSConfig` test should cover the nil case.
2. `docs/manual-test-plan.md` was not updated for the `--skip-verify` reimplementation or the default-Kubernetes-version change; the repo otherwise keeps that plan in sync with behavior changes.
3. The `go.mod` comment and PR body state the fork was carried only for `--skip-verify`; it also carried the rotate-ca `--k8s-endpoint` fix. Worth a one-line correction so the history is accurate.

**Comment by sircthulhu** [comment]
Releasing as alpha does not make much sense, so these changes should only be used for testing. I'll update this PR given that talos v1.14 is released

Do not merge until Talos v1.14 is released.

File: pkg/commands/root.go [hunk 1]
```diff
@@ -110,22 +134,108 @@ func WithClient(action func(context.Context, *client.Client) error, dialOptions
...
+
+	contextName := GlobalArgs.CmdContext
+	if contextName == "" {
+		contextName = cfg.Context
+	}
+
+	configContext, ok := cfg.Contexts[contextName]
+	if !ok {
+		return fmt.Errorf("%w: %q", errContextNotFound, contextName)
+	}
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply pkg/commands/root.go:254]: Already guarded. The nil check landed on main with the fork removal: `root.go` now bails out on `!ok || configContext == nil` before anything dereferences it.

File: pkg/commands/apply.go [hunk 2]
```diff
@@ -797,7 +797,7 @@ func withApplyClientBare(action func(ctx context.Context, c *client.Client) erro
 		return WithClientMaintenance(applyCmdFlags.certFingerprints, action)
 	}
 
-	if GlobalArgs.SkipVerify {
+	if SkipVerify {
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply pkg/commands/apply.go:811]: Checked this against the current tree. For a modelined anchor it does not hold: `resetGlobalArgsBetweenFiles` clears the seeded `127.0.0.1` before the client is built, so the modeline or the talosconfig context supplies the endpoints. For a non-modelined file the seed does survive, but that is not specific to `--skip-verify` — the normal client path applies the same `GlobalArgs.Endpoints` precedence, so both reach the node the same way. The code is byte-identical to main, so nothing here changed it. Worth its own issue rather than this PR if the seeded default is wrong for direct-patch applies.

**Review by lexfrei — DISMISSED** [review]

**Review by lexfrei — DISMISSED** [review]
Re-approving after the two follow-up fixes: the stderr capture helper now restores os.Stderr when the callback aborts, and the test plan names the insecure-apply flag correctly. CI is green on both platforms.

**Review by lexfrei — APPROVED** [review]
Re-approving after the conflict-hint fix and the stale API comments. The v1alpha1 conflict hint now branches on the parsed contract: above v1.13 it still offers the lower pin, at or below it says to drop the duplicated v1alpha1 field, which is the only remedy that applies there. CI green on both platforms.
